src/components/TodosList.js

```diff
@@ -1,22 +1,44 @@
 import styled from 'styled-components';
+import {Button} from './Button';
+
+const TodosList = ({todos, deleteTodo}) => {
+  function onHandleClick(e) {
+    const todoid = e.target.dataset.id;
+    deleteTodo(todoid);
```

Comment:
The whole logic should be at the level your state is managed. Keep your components stupid as possible.